protocol.go

```diff
@@ -83,25 +78,12 @@ func newSecureSession(ctx context.Context, local peer.ID, key ci.PrivKey, insecu
 		return nil, fmt.Errorf("insecure ReadWriter is nil")
 	}
 
-	s.ctx = ctx
 	s.insecure = insecure
 	s.insecureM = msgio.NewReadWriter(insecure)
-	return s, nil
-}
-
-func (s *secureSession) Handshake() error {
-	s.handshakeMu.Lock()
-	defer s.handshakeMu.Unlock()
 
-	if s.handshakeErr != nil {
-		return s.handshakeErr
-	}
-
-	if !s.handshakeDone {
-		s.handshakeErr = s.runHandshake()
-		s.handshakeDone = true
-	}
-	return s.handshakeErr
+	handshakeCtx, cancel := context.WithTimeout(ctx, HandshakeTimeout) // remove
```

Comment:
For record keeping, I've filed an issue about this: https://github.com/libp2p/go-libp2p-secio/issues/17